apps/prairielearn/elements/pl-order-blocks/dag_checker.py

```diff
@@ -221,12 +223,15 @@ def lcs_partial_credit(
     for i in range(len(submission_no_distractors)):
         for j in range(i + 2, len(submission_no_distractors)):
             node1, node2 = submission_no_distractors[i], submission_no_distractors[j]
-            if group_belonging.get(node1) is None or group_belonging.get(
-                node1
-            ) != group_belonging.get(node2):
+            if (
+                node1 is None
+                or node2 is None
+                or group_belonging.get(node1) is None
+                or group_belonging.get(node1) != group_belonging.get(node2)
+            ):
                 continue
             if not all(
-                group_belonging[x] == group_belonging[node1]
+                x is not None and group_belonging[x] == group_belonging[node1]
                 for x in submission_no_distractors[i : j + 1]
```

Comment:
while the `submission` that gets passed as a function argument might have `None` in it, the list `submission_no_distractors` will never have `None` in it. If you add that typing above, then you won't need the None checks here on `node1` and `node2`

Reply:
you also won't need the None check on `x` in the comprehension

Reply:
Resolved in [10c9d37](https://github.com/PrairieLearn/PrairieLearn/pull/11249/commits/10c9d37f2093c05684b81aeb37e5f3e32073056f).